### PR TITLE
refactor!: Provider dependencies standartization

### PR DESCRIPTION
**Pull request by IvayloAndonov**
# Context

As per [this conversation](https://github.com/input-output-hk/cardano-js-sdk/pull/553#discussion_r1041822915), we need a refactoring action to standardize the Provider dependencies in order to enhance the DbSyncProvider parameters to provide it a logger and finally adding the same log line described in the same conversation.
# Proposed Solution

# Important Changes Introduced
- restructure `Asset` directory to be aligned with all other providers
- standardize `DbSyncProvider` dependencies, create interface `DbSynProviderDependencies`
- add debug log on `DbSyncProvider` health check
- align all provider dependencies structure
- add doc blocks for all props and deps interfaces 

**BREAKING CHANGE:**
- all provider constructors are updated to use a standardized form of deps

### CODE REVIEW

**Review by mkazlauskas — APPROVED** [review]
Nice work!

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/DbSyncChainHistoryProvider.ts [hunk 1]
```diff
@@ -42,7 +46,7 @@ export class DbSyncChainHistoryProvider extends DbSyncProvider() implements Chai
     { paginationPageSizeLimit }: ChainHistoryProviderProps,
     { db, cardanoNode, metadataService, logger }: ChainHistoryProviderDependencies
   ) {
-    super(db, cardanoNode);
+    super({ cardanoNode, db, logger });
     this.#logger = logger;
```
Comment (iccicci) [inline packages/cardano-services/src/ChainHistory/DbSyncChainHistory/DbSyncChainHistoryProvider.ts:50]: The `#logger` property can be removed from from `DbSyncChainHistoryProvider` as it now inherits the `logger` property from `DbSyncProvider`.
The same, can be applied to all other `Provider`s extending `DbSyncProvider`.

**Review by iccicci — COMMENTED** [review]
Excellent job! Just a single comment.

File: packages/cardano-services/src/ChainHistory/DbSyncChainHistory/DbSyncChainHistoryProvider.ts (hunk 1 above)
Comment (IvayloAndonov) [reply packages/cardano-services/src/ChainHistory/DbSyncChainHistory/DbSyncChainHistoryProvider.ts]: Good one, applied in 95e990040301ec5bf26de7346cc2b152dae8556e @iccicci

**Review by iccicci — DISMISSED** [review]

**Comment by IvayloAndonov** [comment]
Rebased with master and commits are squashed.

**Review by mkazlauskas — APPROVED** [review]

**Review by rhyslbw — CHANGES_REQUESTED** [review]
Looks good @IvayloAndonov, and good job with the attention to adding docblocks and refactoring for standardisation. There's just a few types that need to be reconsidered though, as they'll be used for other implementations:

https://github.com/input-output-hk/cardano-js-sdk/blob/4b9fd5690fbe7cff0eeae8313536459ec0a0ce7a/packages/cardano-services/src/Asset/DbSyncAssetProvider/types.ts#L4-L31

**Comment by rhyslbw** [comment]
> What is meant here by "to be reconsidered"?

They should not be nested under a `db-sync` scope as they can be used in other contexts. For now you could just leave these interfaces where they were, and only move the `db-sync` models into the new directory.

**Comment by IvayloAndonov** [comment]
> > What is meant here by "to be reconsidered"?
> 
> They should not be nested under a `db-sync` scope as they can be used in other contexts. For now you could just leave these interfaces where they were, and only move the `db-sync` models into the new directory.

Aham, both interfaces are not coupled with `db-sync` at all, only the actual implementations could be, like `DbSyncNftMetadataService`.
So I'll move both types out of `Asset/DbSyncAssetProvider`, good catch!

**Review by rhyslbw — CHANGES_REQUESTED** [review]
https://github.com/input-output-hk/cardano-js-sdk/blob/4b9fd5690fbe7cff0eeae8313536459ec0a0ce7a/packages/cardano-services/src/Asset/DbSyncAssetProvider/types.ts#L33-L50

These are specific to `db-sync`, so best to nest them as you had it originally

**Review by rhyslbw — APPROVED** [review]

File: packages/cardano-services/src/util/DbSyncProvider/DbSyncProvider.ts [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
+  ProviderFailure
+} from '@cardano-sdk/core';
 import { DB_BLOCKS_BEHIND_TOLERANCE, DB_MAX_SAFE_INTEGER, LedgerTipModel, findLedgerTip } from './util';
+import { Logger } from 'ts-log';
 import { Pool } from 'pg';
 
+/**
+ * Properties that are need to create DbSyncProvider
+ */
+export interface DbSynProviderDependencies extends ProviderDependencies {
```
[the remark this answers is not in the document]
Comment (IvayloAndonov) [reply packages/cardano-services/src/util/DbSyncProvider/DbSyncProvider.ts]: The typo is fixed with 48438d21bc630df092ac23cf68d9fed3ad7a49e7 @iadmytro

**Review by iadmytro — APPROVED** [review]

**Comment by IvayloAndonov** [comment]
Commits are squashed and ready for merge once the CI pass.

**Review by rhyslbw — APPROVED** [review]

**Review by iccicci — APPROVED** [review]

**Pull request merged**